sherpa/astro/xspec/__init__.py

```diff
@@ -196,7 +201,7 @@ def __init__(self, filename, name='xstbl', parnames=(),
 
         # make translation table to turn reserved characters into '_'
         bad = string.punctuation+string.whitespace
```

Comment:
Aha - I had forgotten to do an `import string` when I tested. I'd also forgotten about the I-think-should-be-deprecated code saying

```
    try:
        xspec-table-model-code()
    except:
        sherpa-table-model-code()
```